src/bionics.cpp

```diff
@@ -67,10 +69,141 @@ const efftype_id effect_visuals( "visuals" );
+    mvwaddch( win, 2, getmaxx( win ) - 1, LINE_XOXX );
+    mvwaddch( win, getmaxy( win ) - 6, 0, LINE_XXXO );
+    mvwaddch( win, getmaxy( win ) - 6, getmaxx( win ) - 1, LINE_XOXX );
+    wattroff( win, BORDER_COLOR );
+
+    if( empty_list ) {
+        mvwprintz( win, 3, 2, c_ltgray, _( "No bionics installed." ) );
+    } else {
+        // titles
+        static const std::array<std::string, 3> titles = {{
+            _( "Bionic [Size]" ),
+            _( "Condition" ),
+            _( "Usage Cost" )
+        }};
+        std::array<int, 3> pos;
+            pos[0] = 2;
+            pos[1] = std::max( getmaxx( win ) / 2 - utf8_width( titles[1].c_str() ) / 2 ,
+                               pos[0] + utf8_width( titles[0].c_str() ) + 2 );
+            pos[2] = std::max( getmaxx( win ) * 2 / 3 + 7,
+                               pos[1] + utf8_width( titles[1].c_str() ) + 2);
+        for( size_t i = 0; i < std::min( titles.size(), pos.size() ); ++i ) {
+            mvwprintw( win, 3, pos[i], titles[i].c_str() );
+        }
+    }
+    wrefresh( win );
+}
+
+void draw_header( WINDOW *win, const size_t tab_index, const std::string power_string,
+                  const std::string help_key )
+{
+    const std::array<std::string, 3> titles = {{
+        string_format( _( "Body Parts <color_yellow><< %s >></color>" ), bodyparts[tab_index].c_str() ),
+        power_string,
+        string_format( _( "Press '<color_yellow>%s</color>' for help" ), help_key.c_str() )
+    }};
+    // left / center / right aligned elements
+    const std::array<int, 3> pos = {{
+        0,
+        ( getmaxx( win ) - utf8_width( titles[1] ) ) / 2 + 1,
+        ( getmaxx( win ) - utf8_width( titles[2], true ) - 1 )
+    }};
+
+    werase( win );
+    for( size_t i = 0; i < std::min( titles.size(), pos.size() ); ++i ) {
+        fold_and_print( win, 0, pos[i], getmaxx( win ) - 3, c_white, titles[i].c_str() );
+    }
+    wrefresh( win );
+}
+
+const char *power_description( std::string const &id )
+{
+    std::ostringstream power_desc;
+    bool has_previous_text = false;
+    if( bionics[id].power_over_time > 0 && bionics[id].charge_time > 0 ) {
+        power_desc << ( bionics[id].charge_time == 1 ?
+                        string_format( _("%d PU / turn" ), bionics[id].power_over_time ) :
+                        string_format( _("%d PU / %d turns" ), bionics[id].power_over_time,
+                                       bionics[id].charge_time ) );
+        has_previous_text = true;
+    }
+    if( bionics[id].power_activate > 0 && bionics[id].charge_time == 0 ) {
+        if( has_previous_text ) {
+            power_desc << ", ";
+        }
+        power_desc << string_format( _( "%d PU / activation" ),
+                                     bionics[id].power_activate );
+        has_previous_text = true;
+    }
+    if( bionics[id].power_deactivate > 0 && bionics[id].charge_time == 0 ) {
+        if( has_previous_text ) {
+            power_desc << ", ";
+        }
```

Comment:
It could also be wrapped into a function/lambda to reduce duplication and get rid of the conditions:
```c++
const auto separator = []( std::stringstream &s ) {
    return s.tellp() != 0 ? ", " : "";
};
power_desc << separator(power_desc) << "...";
```